### DISCUSSION TITLE
Delete rows inside relation in Adonis 5

### DISCUSSION THREAD

**Discussion by dajpes** [adonisjs/core#1341] [Help] [answered]
Hi, Is it possible to remove data that is inside a relation?

For example in Adonis js 4, you can achieve that like this:

``` Javascript
await user.post().delete()
```
AdonisJs 5:
``` Javascript
await user.related("post").delete() // Error
await user.preload("post").delete() // Error
```

**Comment by iammrsea** [comment]
```javascript
await user.preload('post');
await user.post.delete();
```

**Reply by dajpes** [reply]
I tried that, but an error says that delete method not found.

**Comment by thetutlage** [comment, marked as answer]
Here's how you can access the query builder of a relationship and then just chain the delete method on it.

https://preview.adonisjs.com/guides/model-relations/introduction#access-to-the-direct-query-builder

**Reply by dajpes** [reply]
Hi,

By using this query, it can actually delete all the associate records:

```Javascript
//Typescript
await user?.related("post").query().delete();
```
